File: solutions/csharp/sublist/1/Sublist.cs
using System;
using System.Collections.Generic;

public enum SublistType
{
    Equal,
    Unequal,
    Superlist,
    Sublist
}

public static class Sublist
{
    public static SublistType Classify<T>(List<T> list1, List<T> list2)
        where T : IComparable
    {
        if (list1.Count == list2.Count)
        {
            for (int i =0; i< list1.Count; i++) {
                if (!list1[i].Equals(list2[i])) return SublistType.Unequal;
            }
            return SublistType.Equal;
        } else if (list1.Count < list2.Count)
        {
            if (list1.Count == 0) return SublistType.Sublist;
            for (int j =0; j<= list2.Count-list1.Count; j++)
            {
                if (list1[0].Equals(list2[j]))
                {
                    int i;
                    for (i=1; i< list1.Count; i++)
                    {
                        if (!list1[i].Equals(list2[j+i])) break;
                    }
                    if (i==list1.Count) return SublistType.Sublist;
                }
            }
            return SublistType.Unequal;
        } else
        {
            if (list2.Count == 0) return SublistType.Superlist;
            for (int j =0; j<= list1.Count-list2.Count; j++)
            {
                if (list1[j].Equals(list2[0]))
                {
                    int i;
                    for (i=1; i< list2.Count; i++) {
                        if (!list1[j+i].Equals(list2[i])) break;
                    }
                    if (i==list2.Count) return SublistType.Superlist;
                }
            }
            return SublistType.Unequal;
        }
    }
}

File: solutions/python/sublist/1/sublist.py
"""
This exercise stub and the test suite contain several enumerated constants.

Since Python 2 does not have the enum module, the idiomatic way to write
enumerated constants has traditionally been a NAME assigned to an arbitrary,
but unique value. An integer is traditionally used because it’s memory
efficient.
It is a common practice to export both constants and functions that work with
those constants (ex. the constants in the os, subprocess and re modules).

You can learn more here: https://en.wikipedia.org/wiki/Enumerated_type
"""

# Possible sublist categories.
# Change the values as you see fit.
SUBLIST = 1
SUPERLIST = 2
EQUAL = 3
UNEQUAL = 4


def sublist(list_one, list_two):
    if len(list_one) == 0 and len(list_two) > 0:
        return SUBLIST

    if len(list_one) > 0 and len(list_two) == 0:
        return SUPERLIST 

    if list_one == list_two:
        return EQUAL
    
    if len(list_one) == len(list_two):
        return UNEQUAL

    if len(list_one) <= len(list_two):
        if findsublist(list_one, list_two):  
            return SUBLIST       
    else:
        if findsublist(list_two, list_one):  
            return SUPERLIST 
 
    return UNEQUAL 

def findsublist(list1, list2):
    #print(len(list1),len(list2))
    sp = 0
    j = sp
    i = 0
    while j<len(list2):
        #print(i,j,list1[i],list2[j])
        if list1[i] == list2[j]:
            i += 1
            j += 1
        else:
            sp += 1
            j = sp
            i = 0
        if i >= len(list1):
            return True
    return False

print(findsublist( [3, 4, 5],[0, 1, 2, 3, 4, 5]))        
